worker/caasunitinit/manifold.go

```diff
@@ -112,19 +112,15 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 				return unit.ProviderID(), nil
 			}
 
-			execClient, err := config.NewExecClient(os.Getenv(provider.OperatorNamespaceEnvName))
-			if err != nil {
-				return nil, errors.Trace(err)
-			}
-
 			cfg := Config{
 				Logger:                config.Logger,
 				Application:           applicationTag.Id(),
 				Clock:                 clock,
 				DataDir:               agentConfig.DataDir(),
 				ContainerStartWatcher: client,
 				UnitProviderIDFunc:    unitProviderID,
-				ExecClient:            execClient,
+				NewExecClient:         config.NewExecClient,
+				Namespace:             os.Getenv(provider.OperatorNamespaceEnvName),
```

Comment:
We don't need to pass in namespace if we tweak how we define the NewExecClient param - pass a lamda which calls config.NewExecClient
I'd prefer that to limit the coupling and reduce the number of config params